### ISSUE TITLE
Invalid Binary Target for OpenSSL V3 on CentOS Stream 9

### ISSUE THREAD

**Issue by ron-bytes** [prisma/prisma#11677]
### Bug description

Prisma CLI fails to initialize when deploying a new or existing project to a CentOS Stream 9 server. The error references an unknown binary target for openssl 3.0.x.

```
# npx prisma init
  prisma:engines binaries to download libquery-engine, migration-engine, introspection-engine, prisma-fmt +0ms
  prisma:download file /root/hello-prisma/node_modules/@prisma/engines/libquery_engine-rhel-openssl-3.0.x.so.node does not exist and must be downloaded +0ms
Error: Error: Unknown binaryTarget rhel-openssl-3.0.x and no custom binaries were provided
    at /root/hello-prisma/node_modules/@prisma/engines/dist/index.js:21277:15
    at async Promise.all (index 0)
    at async /root/hello-prisma/node_modules/@prisma/engines/dist/index.js:6621:29
```

The same error is received when trying to initialize the new project directory or generating artifacts (prisma client) from an existing project.

OpenSSL v3.0.1 is confirmed to be installed.

```
# openssl version
OpenSSL 3.0.1 14 Dec 2021 (Library: OpenSSL 3.0.1 14 Dec 2021)
# cat /etc/redhat-release
CentOS Stream release 9
```

Running `npx prisma` completes successfully and shows help output.

### How to reproduce

1. Start with an updated CentOS Stream 9 server with Node installed.
2. Create a new project. `mkdir hello-prisma`
3. Enter the directory. `cd hello-prisma`
4. Initialize the project. `npm init -y`
5. Install prisma. `npm install prisma --save-dev`
6. Test prisma to make sure it is working. `npx prisma`
7. Initialize prisma. `npx prisma init`
8. The error is now received.

### Expected behavior

Prisma CLI should work as expected correctly generate the project files.

### Prisma information

N/A as the error prevents a project from getting this far.

### Environment & setup

- OS: CentOS Stream 9
- Database: MySQL
- Node.js version: v16.10.0

### Prisma Version

Prisma version flag fails to complete. I can confirm prisma was installed from NPM as detailed in the quickstart.

```
# npx prisma -v
  prisma:engines binaries to download libquery-engine, migration-engine, introspection-engine, prisma-fmt +0ms
  prisma:download file /root/hello-prisma/node_modules/@prisma/engines/libquery_engine-rhel-openssl-3.0.x.so.node does not exist and must be downloaded +0ms
Error: Error: Unknown binaryTarget rhel-openssl-3.0.x and no custom binaries were provided
    at /root/hello-prisma/node_modules/@prisma/engines/dist/index.js:21277:15
    at async Promise.all (index 0)
    at async /root/hello-prisma/node_modules/@prisma/engines/dist/index.js:6621:29
```

**Comment by pedrovsg** [comment]
I have the issue for debian
`Error: Unknown binaryTarget debian-openssl-3.0.x and no custom binaries were provided`

**Comment by mbaroukh** [comment]
Is there a method to force prisma to use a different version ?  
I did something not really clean to force prisma to think I have opensssl 1.1.1 and it works for now so it might be a workaround in the meantime.

**Comment by matthewmueller** [comment]
Note: We changed this to a feature because although it looks like a bug, we don't publish binaries for openssl 3.0 yet because we still need to add support for it.

**Comment by janpio** [comment]
The feature request to support OpenSSL 3 is here: https://github.com/prisma/prisma/issues/11356 From a user perspective, Prisma not working in their environment is a defect that we track as a bug. When the feature request is resolved, we can also resolve the bug issue tracking the defect.

**Comment by CDFN** [comment]
If somebody would like to build binaries for themeselves instead of waiting, just clone https://github.com/prisma/prisma-engines, run `cargo build --release` (`pkg-config` lib required in addition to rust toolcahin, at least in case of Ubuntu 22.04, not sure about others) and set those envs:
```bash
export PRISMA_QUERY_ENGINE_LIBRARY=/home/cdfn/dev/prisma-engines/target/release/libquery_engine.so
export PRISMA_QUERY_ENGINE_BINARY=/home/cdfn/dev/prisma-engines/target/release/query-engine
export PRISMA_MIGRATION_ENGINE_BINARY=/home/cdfn/dev/prisma-engines/target/release/migration-engine
export PRISMA_INTROSPECTION_ENGINE_BINARY=/home/cdfn/dev/prisma-engines/target/release/introspection-engine
export PRISMA_FMT_BINARY=/home/cdfn/dev/prisma-engines/target/release/prisma-fmt
export PRISMA_CLI_QUERY_ENGINE_TYPE=binary # you may not need this, I had weird error with lib not being able to load
export PRISMA_QUERY_ENGINE_TYPE=binary # you may not need this, I had weird error with lib not being able to load
```
Additionaly, you need to add in your `schema.prisma`:
```diff
 generator client {
    provider      = "prisma-client-js"
+   binaryTargets = ["darwin"]
 }

```
For me this setup works perfectly well.

**Comment by fomenkoyegor** [comment]
wtf !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! f*  prisma developers

**Comment by CDFN** [comment]
Today Ubuntu 22.04 dropped. It uses OpenSSL V3 by default. Is there any progress on this or we're forced to downgrade?

**Comment by janpio** [comment]
This is fixed and will ship with 3.13.0:
https://github.com/prisma/prisma/issues/11356#issuecomment-1103977740
That version will be release next Tuesday, and this comment includes instructions how you can temporarily use the development version of Prisma to confirm it works: https://github.com/prisma/prisma/issues/11356#issuecomment-1106208683
